fix(files): now uses filename on filestore to detect mime

hypeJunction opened the pull request:
For ElggFile instances ElggFile::detectMimeType was using filename metadata,
which was not a readable file. Now normalizes filename to a filestore path,
and ensures that the file is readable before calling finfo_\* and mime_*

Fixes #8846

hypeJunction committed:
fix(files): now uses filename on filestore to detect mime

For ElggFile instances ElggFile::detectMimeType was using filename metadata,
which was not a readable file. Now normalizes filename to a filestore path,
and ensures that the file is readable before calling finfo_* and mime_*

Fixes #8846

hypeJunction committed:
fix(files): $file is not always an object with originalfilename property

$file can be a string in ElggFile::detectMimetype, which was producing log errors.
Now using actual ElggFile instance to get the originalfilename metadata, or
basename() to get the filename from a path.

Refs #8846

mrclay commented on engine/classes/ElggFile.php:
```diff
@@ -154,9 +154,10 @@ public function detectMimeType($file = null, $default = null) {
 			$mime = mime_content_type($file);
 		}
 
+		$original_filename = isset($this) ? $this->originalfilename : basename($file);
```
Can you add a `@note` in the phpdoc that we're calling this both dynamically and statically?

mrclay commented:
Besides that, looks OK to me.

hypeJunction committed:
chore(files): document the static use of ElggFile::detectMimeType method

Adds a note about the static and dynamic use of ElggFile::detectMimeType method

juho-jaakkola commented:
LGTM
